### ISSUE TITLE
Immutable builder from public all arg constructor

### ISSUE THREAD

**Issue by crumley** [immutables/immutables#433]
This requests comes from a desire to use Immutables with java code/classes generated from an avro schema but would be applicable for anyone who wishes to use Immutables with a class that has a public all argument constructor. 

For example given the pojo:
```
public class Pojo {
   public Pojo(int a, int b) { ... }
}
```

I'd like to use Immutables to generate a builder than eventually calls the constructor Pojo(int, int) to construct the object. I can do this today by creating a factory builder:

```
public class SomeClass {
   @Builder.Factory
   public static Pojo pojo(int a, int b) {
       return new Pojo(a, b);
   }
}
``` 

Then I can use PojoBuilder to construct new instances which is more convenient than using the constructor itself (imagine a constructor with 20 arguments, half of which are optional). But writing and maintaining this static function is tedious. It would be great if the static function is not required and instead @Builder.Factory could take any class and make a builder for its constructor.

@Value.Include provides a mechanism for using Immutables with source you do not control but is limited to abstract and interface classes, would be nice to be able to do something similar with a builder and any classes constructor.

**Comment by elucash** [comment]
Created annotation `Builder.Constructor` to annotated constructors. The functionality is not well tested, but should work of for many cases. You can try it out in snapshots or building Immutables from source. Will be released with the next release. Thank you!

**Comment by elucash** [comment]
There's also `@Builder.Include` to auto include constructors and static methods from external classes and turn them into builders

**Comment by crumley** [comment]
Builder.Include looks perfect! Thank you.

**Comment by elucash** [comment]
released in 2.3
